celery/worker/request.py

```diff
@@ -152,7 +152,8 @@ def __init__(self, message, on_ack=noop,
             'delivery_info': {
                 'exchange': delivery_info.get('exchange'),
                 'routing_key': delivery_info.get('routing_key'),
-                'priority': delivery_info.get('priority'),
+                'priority': delivery_info.get('priority',
+                                              properties.get('priority')),
```

Comment:
Seems I made a mistake in the virtual kombu transports here, as they set this in `delivery_info`.
Since this is for the 4.0 release we can just as well fix it properly, and read it from properties directly.